pkg/actors/actors.go

```diff
@@ -150,6 +151,7 @@ func NewActors(
 	features []configuration.FeatureSpec,
 	resiliency resiliency.Provider,
 	stateStoreName string,
+	workflowEngineChannel channel.AppChannel,
```

Comment:
Is your concern just about constructor arguments? If so, yes, I think I can refactor the initialization path. However, making it more generic will bloat the changes to the actor runtime. My previous iteration was more generic such that actors didn't know anything about workflows, but I got pushback on this because of the new concepts that were introduced. I'm a little worried that I won't be able to make everyone happy with this PR.

Reply:
So I can make it so the actor system doesn't need to know about the workflow engine. However, I don't think I can remove the need for a constructor argument. This constructor returns an `actors.Actors` interface. However, the `AppChannel` is an implementation detail that wouldn't make sense to add to the `actors.Actors` interface. On top of that, all the actor initialization happens inside the `Init()` method, meaning all internal actors need to be registered before the runtime is initialized. I looked at this from a few different angles, and there really isn't any cleaner way to inject internal actors via the `AppChannel` abstraction except via this constructor.